datashader/glyphs/quadmesh.py

```diff
@@ -215,7 +215,7 @@ def extend(aggs, xr_ds, vt, bounds, x_breaks=None, y_breaks=None):
             else:
                 do_extend = extend_cpu
 
-            do_extend(xs, ys, *aggs_and_cols)
+            do_extend(xs, ys, tuple(aggs[0].shape), *aggs_and_cols)
```

Comment:
Technically `aggs_and_cols` contains the arrays we're accumulating into, so we could obtain the shape from there. However since we have `ast` hacks to unpack these I'm not sure there's a clean way to make that happen.

Reply:
I tried to do that as the first thing,  but couldn't figure out an easy way to extract it inside a numba function, as it will throw a KeyError when trying to unpack it:

``` python-traceback
Traceback (most recent call last):
  File "/home/shh/.local/conda/envs/holoviz/lib/python3.13/site-packages/toolz/functoolz.py", line 457, in memof
    return cache[k]
           ~~~~~^^^
KeyError: (<datashader.glyphs.quadmesh.QuadMeshRectilinear object at 0x7f9a95d33cb0>, CPUDispatcher(<function LinearAxis.mapper at 0x7f9a95dd5b20>), CPUDispatcher(<function LinearAxis.mapper at 0x7f9a95dd5b20>), <function make_info.<locals>.info at 0x7f9b983344a0>, CPUDispatcher(<function append at 0x7f9a95cddf80>), False, None)
```